### PR TITLE
updated libxxhash.pc generation

### PR DESCRIPTION
**Pull request by Cyan4973**
can control libdir and includedir

fix #437

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -413,54 +411,82 @@ INSTALL_PROGRAM ?= $(INSTALL)
...
 
 
+PCLIBDIR ?= $(shell echo "$(LIBDIR)"     | $(SED) -n $(SED_ERE_OPT) -e "s@^$(EXEC_PREFIX)(/|$$)@@p")
+PCINCDIR ?= $(shell echo "$(INCLUDEDIR)" | $(SED) -n $(SED_ERE_OPT) -e "s@^$(PREFIX)(/|$$)@@p")
+
+ifeq (,$(PCLIBDIR))
+# Additional prefix check is required, since the empty string is technically a
+# valid PCLIBDIR
+ifeq (,$(shell echo "$(LIBDIR)" | $(SED) -n $(SED_ERE_OPT) -e "\\@^$(EXEC_PREFIX)(/|$$)@ p"))
+$(error configured libdir ($(LIBDIR)) is outside of prefix ($(PREFIX)), can't generate pkg-config file)
```
Comment (ellert) [inline Makefile]: ... is outside of exec_prefix ($(EXEC_PREFIX))

File: Makefile [hunk 2]
```diff
@@ -413,54 +411,82 @@ INSTALL_PROGRAM ?= $(INSTALL)
...
+endif
+endif
+
+ifeq (,$(PCINCDIR))
+# Additional prefix check is required, since the empty string is technically a
+# valid PCINCDIR
+ifeq (,$(shell echo "$(INCLUDEDIR)" | $(SED) -n $(SED_ERE_OPT) -e "\\@^$(PREFIX)(/|$$)@ p"))
+$(error configured includedir ($(INCLUDEDIR)) is outside of exec_prefix ($(EXEC_PREFIX)), can't generate pkg-config file)
```
Comment (ellert) [inline Makefile]: ... is outside of prefix ($(PREFIX))

File: libxxhash.pc.in [hunk 3]
```diff
@@ -1,10 +1,11 @@
 #   xxHash - Extremely fast hash algorithm
+#   Copyright (C) 2012-2020, Yann Collet, Facebook
 #   BSD 2-Clause License (https://www.opensource.org/licenses/bsd-license.php)
 
 prefix=@PREFIX@
 exec_prefix=${prefix}
```
Comment (ellert) [inline libxxhash.pc.in]: In theoty, someone cound define EXEC_PREFIX different than PREFIX. The libdir in the .pc file would then evaluate wrong. (Not a common case though).

**Comment by Cyan4973** [comment]
Good catch @ellert !
I believe latest update should fix those issues.

File: Makefile [hunk 4]
```diff
@@ -413,54 +411,84 @@ INSTALL_PROGRAM ?= $(INSTALL)
 INSTALL_DATA    ?= $(INSTALL) -m 644
 
 
+PCLIBDIR ?= $(shell echo "$(LIBDIR)"     | $(SED) -n $(SED_ERE_OPT) -e "s@^$(EXEC_PREFIX)(/|$$)@@p")
+PCINCDIR ?= $(shell echo "$(INCLUDEDIR)" | $(SED) -n $(SED_ERE_OPT) -e "s@^$(PREFIX)(/|$$)@@p")
+PCEXECDIR?= $(if $(filter $(PREFIX),$(EXEC_PREFIX)), $$\{prefix\}, $(EXEC_PREFIX))
```
Comment (ellert) [inline Makefile]:
```
-PCEXECDIR?= $(if $(filter $(PREFIX),$(EXEC_PREFIX)), $$\{prefix\}, $(EXEC_PREFIX))
+PCEXECDIR ?= $(if $(filter $(PREFIX),$(EXEC_PREFIX)),$$\{prefix\},$(EXEC_PREFIX))
```
The .pc file says:
exec_prefix= ${prefix}
with an extra space. This doesn't make it broken or anything, so feel free to ignore this...

**Pull request merged**